doc/neps/nep-0038-SIMD-optimizations.rst

```diff
@@ -223,15 +268,28 @@ implementing and maintaining that platform's loop code.
 Discussion
 ----------
 
-.. note::
-    Will include a summary of the discussion once the NEP is published
+Most of the discussion took place on the PR `gh-15228`_ to accecpt this NEP.
+Discussion on the mailing list mentioned `VOLK`_ which was added to
+the section on related work. The question of maintainability also was raised
+both on the mailing list and in `gh-15228`_ but not resolved. The open
+questions are:
+- If contributors want to leverage a specific SIMD instruction, will they be
+  expected to add software implementation of this instruction for all other
+  architectures too?
```

Comment:
I'm not sure this was the question. It makes sense (answer is yes I think or things will break), but IIRC the discussion was mainly about adding a ufunc implementation for (e.g.) `x86_64` and being expected to deal with ARM/PPC/etc. yes or no; not a new missing SIMD instruction.

EDIT: I checked the mailing list, this was indeed the question there. I'll reply, because I don't think the answer was correct.

Reply:
@rgommers is there something to change here?

Reply:
Yes. I propose to say as a guideline:

If a contributor wants to use a platform-specific SIMD instruction that is not yet supported as a universal intrinsic, then:
1. If should be added as a universal intrinsic for all platforms
2. If it _does not have_ an equivalent instruction on other platforms (e.g. `_mm512_mask_i32gather_ps` in AVX512), then no universal intrinsic should be added and a platform-specific implementation of the ufunc should be written instead.

We expect (2) to be the exception. The contributor and maintainers should consider whether that single-platform intrinsic is worth it compared to using the best available universal intrinsic based implementation.

Reply:
> 2. If it does not have an equivalent instruction on other platforms,
  
almost any intrinsic can be emulated if it doesn't exist. if you consider the emulated intrinsics in #13516 can't be an example, well, in that case, you should consider @r-devulap emulated intrinsics for (AVX2 & FMA3) in [simd.inc.src](https://github.com/numpy/numpy/blob/master/numpy/core/src/umath/simd.inc.src) an example, in other words, we already have universal intrinsics but only for (AVX512, AVX2), we just going to expand it more  :).

Here some examples instead of looking in the code:
### case: get_exponent:
**AVX512 version**:
https://github.com/numpy/numpy/blob/1f9ab28a9ee5a0cda956da35d0c77a3a4c4bcd55/numpy/core/src/umath/simd.inc.src#L1648-L1651

**(AVX2) version(emulated)**:
https://github.com/numpy/numpy/blob/1f9ab28a9ee5a0cda956da35d0c77a3a4c4bcd55/numpy/core/src/umath/simd.inc.src#L1417-L1441

### case: get_mantissa
**AVX512 version**:
https://github.com/numpy/numpy/blob/1f9ab28a9ee5a0cda956da35d0c77a3a4c4bcd55/numpy/core/src/umath/simd.inc.src#L1654-L1657
**(AVX2) version(emulated)**:
https://github.com/numpy/numpy/blob/1f9ab28a9ee5a0cda956da35d0c77a3a4c4bcd55/numpy/core/src/umath/simd.inc.src#L1444-L1467

Reply:
I think that proves @rgommer's point (if you replace "ufunc" with "helper function": the cross-platform universal intrinsics do not exist for `get_mantissa`, so platform-specific helper functions are used instead. We do not define a universal intrinsic `npy_get_mantissa` if it does not exist on all platforms. So I will add the comment with the helper function caveat